src/utils/request.ts

```diff
@@ -129,3 +133,20 @@ export function getRequestURL(
   const path = getRequestPath(event);
   return new URL(path, `${protocol}://${host}`);
 }
+
+export function setOriginalUrlPath(event: H3Event, url: string) {
+  // eslint-disable-next-line @typescript-eslint/no-extra-semi
+  (event.node.req as any).originalUrlPath = url;
+}
+
+export function getOriginalUrlPath(event: H3Event) {
+  return (event.node.req as any).originalUrlPath as string;
+}
+
+export function setUrlPath(event: H3Event, url: string) {
+  event.node.req.url = url;
+}
+
+export function getUrlPath(event: H3Event) {
+  return event.node.req.url || "/";
```

Comment:
Why we need this and setter?

Reply:
It is used in many places and end-users might be tempted to use it: https://github.com/unjs/h3/blob/091f32602c8133d355d9c9ed784e55ca53099dec/src/utils/route.ts#L12

Reply:
Why not advice `event.path` (which is essentially `req.url` + some security fixes)

Reply:
The API is helper based, `getPath(event)`  would be more consistent than `event.path` for end users.
The setter is probably internal use only,  and it was needed for the fetch adapter.

Reply:
We can directly set `_*` internals from within other internals such as fetch adapter if there is no need for a reusable normalization util.

Reply:
What should we do then ? Remove the setter and expose `getPath(event)` ?